src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -128,37 +226,89 @@ public Assembly LoadFromStream(Stream assembly, Stream assemblySymbols)
             {
                 throw new BadImageFormatException(SR.BadImageFormat_BadILFormat);
             }
+#if FEATURE_COLLECTIBLE_ALC
+            lock (unloadLock)
+            {
+                VerifyIsAlive();
+#endif
+                int iAssemblyStreamLength = (int) assembly.Length;
+                int iSymbolLength = 0;
 
-            int iAssemblyStreamLength = (int)assembly.Length;
-            int iSymbolLength = 0;
+                // Allocate the byte[] to hold the assembly
+                byte[] arrAssembly = new byte[iAssemblyStreamLength];
 
-            // Allocate the byte[] to hold the assembly
-            byte[] arrAssembly = new byte[iAssemblyStreamLength];
+                // Copy the assembly to the byte array
+                assembly.Read(arrAssembly, 0, iAssemblyStreamLength);
 
-            // Copy the assembly to the byte array
-            assembly.Read(arrAssembly, 0, iAssemblyStreamLength);
+                // Get the symbol stream in byte[] if provided
+                byte[] arrSymbols = null;
+                if (assemblySymbols != null)
+                {
+                    iSymbolLength = (int) assemblySymbols.Length;
+                    arrSymbols = new byte[iSymbolLength];
+
+                    assemblySymbols.Read(arrSymbols, 0, iSymbolLength);
+                }
 
-            // Get the symbol stream in byte[] if provided
-            byte[] arrSymbols = null;
-            if (assemblySymbols != null)
+                RuntimeAssembly loadedAssembly = null;
+                unsafe
+                {
+                    fixed (byte* ptrAssembly = arrAssembly, ptrSymbols = arrSymbols)
+                    {
+                        LoadFromStream(m_pNativeAssemblyLoadContext, new IntPtr(ptrAssembly), iAssemblyStreamLength,
+                            new IntPtr(ptrSymbols), iSymbolLength, JitHelpers.GetObjectHandleOnStack(ref loadedAssembly));
+                    }
+                }
+                return loadedAssembly;
+#if FEATURE_COLLECTIBLE_ALC
+            }
+#endif
+        }
+
+        public void Unload()
```

Comment:
I forgot what contract for this method we agreed on: Is it required to call this method to unload the stuff? Or is the ALC going to auto-unload itself too when these are no references to it?

It may be worth a comment.